petab/core.py

```diff
@@ -900,3 +900,99 @@ def measurements_have_replicates(measurement_df: pd.DataFrame) -> bool:
             measurement_df,
             ['observableId', 'simulationConditionId',
              'preequilibrationConditionId', 'time'])).size().values - 1)
+
+
+def flatten_timepoint_specific_output_overrides(
+        petab_problem: Problem) -> None:
+    """If the PEtab problem definition has timepoint-specific
+    observableParameters or noiseParameters for the same observable, replace
+    those by replicating the respective observable.
+
+    This is a helper function for some tools which may not support such
+    timepoint-specific mappings.
+
+    Arguments:
+        petab_problem:
+            PEtab problem to work on
+    """
+
+    # Create empty df -> to be filled with replicate-specific observables
+    df_new = pd.DataFrame()
+
+    # Get observableId, preequilibrationConditionId
+    # and simulationConditionId columns in measurement df
+    df = petab_problem.measurement_df[
+        ["observableId",
+         "preequilibrationConditionId",
+         "simulationConditionId"]
+    ]
+    # Get unique combinations of observableId, preequilibrationConditionId
+    # and simulationConditionId
+    df_unique_values = df.drop_duplicates()
+
+    # Loop over each unique combination
+    for nrow in range(len(df_unique_values.index)):
+        df = petab_problem.measurement_df.loc[
+            (petab_problem.measurement_df['observableId'] ==
+             df_unique_values.loc[nrow, "observableId"])
+            & (petab_problem.measurement_df['preequilibrationConditionId'] <=
```

Comment:
now observableId is checked with ==, but why are the others still checked with <=?